terraform/modules/cf_hosting/cloudfront.tf

```diff
@@ -18,7 +23,94 @@ resource "aws_cloudfront_distribution" "s3_distribution" {
 #   aliases = ["mysite.example.com", "yoursite.example.com"]
 
 	default_cache_behavior {
-		allowed_methods  = ["DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT"]
+		allowed_methods  = ["GET", "HEAD", "OPTIONS"]
+		cached_methods   = ["GET", "HEAD"]
+		target_origin_id = "${var.tags.org_name}-${var.tags.application_name}-${var.tags.environment_name}-orgin"
+
+		forwarded_values {
+			query_string = false
+			headers      = ["CloudFront-Viewer-Country", "x-nice-allowed-ip"]
+
+			cookies {
+				forward = "none"
+			}
+		}
+		lambda_function_association {
+			event_type = "origin-request"
+			lambda_arn = var.origin_request_edge_lambda_qualified_arn
+		}
+		lambda_function_association {
+			event_type = "origin-response"
+			lambda_arn = var.origin_repsonse_edge_lambda_qualified_arn
+		}
+		lambda_function_association {
+			event_type = "viewer-request"
+			lambda_arn = var.viewer_request_edge_lambda_qualified_arn
+		}
+
+		compress               = true
+		viewer_protocol_policy = "redirect-to-https"
+	}
+
+	ordered_cache_behavior {
+		path_pattern = "*.js"
+
+		allowed_methods  = ["GET", "HEAD", "OPTIONS"]
+		cached_methods   = ["GET", "HEAD"]
+		target_origin_id = "${var.tags.org_name}-${var.tags.application_name}-${var.tags.environment_name}-orgin"
+
+		forwarded_values {
+			query_string = false
+			headers      = ["CloudFront-Viewer-Country"]
+
+			cookies {
+				forward = "none"
+			}
+		}
+		lambda_function_association {
+			event_type = "origin-request"
```

Comment:
Do we even need to run the origin request and response lambdas on static files that aren't HTML?

Reply:
yeah, we need the response headers I guess as they add the security info (do we?). But agree as long as we don't mind the static files not being restricted that its fine (and prob a better option as less cost)